src/runtime.js

```diff
@@ -323,20 +348,26 @@ let tornado = {
        return typeof val === 'function';
      },
 
-     /**
-      * Determine Tornado truthiness
-      * @param {*} val The value in question
-      * @return {Boolean}
-      */
-     isTruthy(val) {
-       if (val === 0) {
-         return true;
-       }
-       if (Array.isArray(val) && !val.length) {
-         return false;
-       }
-       return !!val;
-     }
+    /**
+     * Determine Tornado truthiness
+     * @param {*} val The value in question
+     * @return {Boolean}
+     */
+    isTruthy(val) {
+      if (val === 0) {
+        return true;
+      }
+      if (Array.isArray(val) && !val.length) {
+        return false;
+      }
+      return !!val;
+    }
+  },
+
+  attrFilters: {
+    'a[href]': function(val) {
+      return val.replace(/^javascript:.*/, '');
```

Comment:
What I've seen have all been whitelists.   Restrict urls to be one of :
    //    - full urls with https? ftp and mailto protocols (javascript: is not allowed)
    //    - relative paths (/foo)
    //    - hash fragments (#foo) and querystrings (?a=b)

and disallow everything else.

there's a regex for this if you need it.